projects/plugins/boost/app/assets/src/js/features/critical-css/lib/generate-critical-css.ts

```diff
@@ -89,9 +88,6 @@ async function generateCriticalCss(
 	signal: AbortSignal
 ) {
 	try {
-		// Load Critical CSS gen library if not already loaded.
-		await loadCriticalCssLibrary();
-
```

Comment:
The function previously triggered a "nice" error explaining that a dependency is missing. Before we execute `generateCriticalCss` we have to check if critical css generator exists, and if it doesn't - fail gracefully.

Reply:
Can you tell me what error was that? I can't find it.

There's a separate ticket to check if the library is present and the correct thing just before using it https://github.com/Automattic/jetpack/issues/37091

Reply:
I meant this - it tried to load the script and it displays a message that it wasn't able to:

```
scriptTag.addEventListener( 'error', () =>
			reject(
				new Error(
					sprintf(
						/* translators: %s refers to Critical CSS Gen library script url. */
						__( 'Failed to load Critical CSS library at %s', 'jetpack-boost' ),
						scriptUrl
					)
				)
			)
		);
		const timeout = setTimeout( () => {
			reject(
				new Error(
					sprintf(
						/* translators: %s refers to Critical CSS Gen library script url. */
						__( 'Timeout while loading Critical CSS library at %s', 'jetpack-boost' ),
						scriptUrl
					)
				)
			);
		}, 60000 );
		```
		
I think we need something similar for "Critical CSS is missing a dependency" but instead of listening on scriptTag, we look for `window.[.....]` value

Reply:
> I think we need something similar for "Critical CSS is missing a dependency" but instead of listening on scriptTag, we look for `window.[.....]` value

I agree yes. Though I'm not sure it should be here. A separate PR makes more sense to me.